### PR TITLE
Feature/error messages partial fix for #10

### PR DESCRIPTION
**Pull request by Leopard2A5**
Partial fix for #10 
Sorry it's a bit lengthy, but it wasn't easy to break this up!

### CODE REVIEW

File: validator/src/contains.rs [hunk 1]
```diff
@@ -40,8 +40,12 @@ impl<'a, S> Contains for &'a HashMap<String, S> {
 /// Validates whether the value contains the needle
 /// The value needs to implement the Contains trait, which is implement on String, str and Hashmap<String>
 /// by default.
-pub fn validate_contains<T: Contains>(val: T, needle: &str) -> bool {
-    val.has_element(needle)
+pub fn validate_contains<T: Contains>(val: T, needle: &str) -> Result<(), String> {
+    if val.has_element(needle) {
+        Ok(())
+    } else {
+        Err(format!("must contain '{}'", needle))
```
Comment (Keats) [inline validator/src/contains.rs]: I don't want the messages to be in the `validator` crate, they should be in the `validator_derive`.
Validator fns should just return a bool

File: validator/src/types.rs [hunk 2]
```diff
@@ -49,6 +59,39 @@ impl std::error::Error for Errors {
     }
 }
 
+#[derive(Debug, PartialEq)]
+pub struct Error {
```
Comment (Keats) [inline validator/src/types.rs]: `Error` -> `ValidationError`?

File: validator/src/types.rs [hunk 3]
```diff
@@ -49,6 +59,39 @@ impl std::error::Error for Errors {
...
+    validator: String,
```
Comment (Keats) [inline validator/src/types.rs]: `validator` -> `code`

File: validator/src/types.rs (hunk 2 above)
Comment (Leopard2A5) [inline validator/src/types.rs]: I kept with your approach, the Errors struct isn't called ValidationErrors, so why add the prefix for the singular struct?

**Comment by Leopard2A5** [comment]
Well if you want to implement it yourself just say it and don't merge this PR.

File: validator/src/types.rs (hunk 2 above)
Comment (Keats) [inline validator/src/types.rs]: It clashes with `std::error::Error`, at least in my mind. But after thinking about it, it's not that bad

**Comment by Keats** [comment]
Sorry about that :(

**Comment by Keats** [comment]
Closing in favour of #27

**Pull request closed without merging**